presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSource.java

```diff
@@ -105,7 +106,7 @@ public ParquetPageSource(
             }
             else {
                 String columnName = useParquetColumnNames ? name : fileSchema.getFields().get(column.getHiveColumnIndex()).getName();
-                fieldsBuilder.add(constructField(type, messageColumnIO.getChild(columnName)));
+                fieldsBuilder.add(constructField(type, getColumnIOByName(messageColumnIO, columnName)));
```

Comment:
`getColumnIOByName` can return `null`, and that will cause a NPE here in `add` call. If `getColumnIOByName` returns `null` it means a column with given name couldn't be found, and we get the column name from the Parquet file schema, so I feel like that's a serious error condition and instead of returning a null we should throw.

Reply:
I don't know which one is better. I feel that, in the most case, if the field missed in Parquet, it seems OK to return `null`. For example, the `getParquetType` may also return `null`. How do you think?